### ISSUE TITLE
Travis log "command foo exited with <code>" blends into subsequent step

### ISSUE THREAD

**Issue by edmorley** [travis-ci/travis-ci#6281]
tl;dr: Some newline tweaks to the travis log output would make the logs much more easy to read.

This is probably best described with an example. In this log:
https://travis-ci.org/mozilla/treeherder/jobs/143190163

The output currently looks like:

```
> 169: $ pip install ...
  234: $ grunt checkjs
  235: Running "eslint:target" (eslint) task
  236: 
  237: 
  238: Done, without errors.
  239: 
  240: 
  241: The command "grunt checkjs" exited with 0.
  242: $ python lints/queuelint.py
  243: 
  244: 
  245: The command "python lints/queuelint.py" exited with 0.
  246: $ flake8 --show-source
  ...and so on.
```

The problem is that the `The command "foo" exited with 0.` lines:
- have leading newlines (x2), which visually separate them from the command to which they were referring.
- have no newline after them, to separate them from the _next_ command

This is exacerbated since those lines are bright green, so appear almost like headings.

More useful would be output like:

```
> 169: $ pip install ...
  234: $ grunt checkjs
  235: Running "eslint:target" (eslint) task
  236: 
  237: 
  238: Done, without errors.
  239: The command "grunt checkjs" exited with 0.
  240: 
  241: 
  242: $ python lints/queuelint.py
  243: The command "python lints/queuelint.py" exited with 0.
  244: 
  245: 
  246: $ flake8 --show-source
  ...and so on.
```

The raw log for the above shows at least where some of the newlines are coming from:

```
[32;1mThe command "grunt checkjs" exited with 0.[0m
travis_time:start:0b7f0207
[0K$ python lints/queuelint.py

travis_time:end:0b7f0207:start=1467931808972794338,finish=1467931808992599433,duration=19805095
[0K
[32;1mThe command "python lints/queuelint.py" exited with 0.[0m
travis_time:start:0fe0e908
[0K$ flake8 --show-source
```

Many thanks!

**Comment by edmorley** [comment]
There seems to be a bit of inconsistency in the various output functions, as to whether to assume the previous output had a trailing newline or not. ie:
(see https://github.com/travis-ci/travis-build/blob/bbe7c12b6f2c8bdc6cd9a7d3e839a729048648ae/lib/travis/build/templates/header.sh)

For the output of the command that was run:
`echo "\$ ${display:-$cmd}"`

For `travis_time_start()`:
`echo -en "travis_time:start:$travis_timer_id\r${ANSI_CLEAR}"`

For `travis_time_finish()`:
`echo -en "\ntravis_time:end:$travis_timer_id:start=$travis_start_time,finish=$travis_end_time,duration=$duration\r${ANSI_CLEAR}"`

For `travis_result()`:
`echo -e "\n${ANSI_GREEN}The command \"$TRAVIS_CMD\" exited with $result.${ANSI_RESET}"`
or
`echo -e "\n${ANSI_RED}The command \"$TRAVIS_CMD\" exited with $result.${ANSI_RESET}"`

**Comment by edmorley** [comment]
This is still applicable
